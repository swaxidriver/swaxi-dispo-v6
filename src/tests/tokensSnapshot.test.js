import { promises as fs } from 'fs';
import path from 'path';

describe('Design Tokens Stability', () => {
  test('tokens.json snapshot remains stable', async () => {
    const tokensPath = path.join(process.cwd(), 'src', 'tokens.json');
    
    // Read the generated tokens.json
    const tokensContent = await fs.readFile(tokensPath, 'utf8');
    const tokens = JSON.parse(tokensContent);
    
    // Remove volatile metadata that changes on each generation
    const stableTokens = {
      ...tokens,
      meta: {
        source: tokens.meta.source,
        version: tokens.meta.version
        // Exclude 'generated' timestamp as it changes each time
      }
    };
    
    // Snapshot test to ensure tokens structure and values remain stable
    expect(stableTokens).toMatchSnapshot();
  });
  
  test('tokens.json contains required token categories', async () => {
    const tokensPath = path.join(process.cwd(), 'src', 'tokens.json');
    const tokensContent = await fs.readFile(tokensPath, 'utf8');
    const tokens = JSON.parse(tokensContent);
    
    // Verify structure
    expect(tokens).toHaveProperty('light');
    expect(tokens).toHaveProperty('dark');
    expect(tokens).toHaveProperty('meta');
    
    // Verify light theme has required categories
    expect(tokens.light).toHaveProperty('colors');
    expect(tokens.light).toHaveProperty('typography');
    expect(tokens.light).toHaveProperty('radii');
    expect(tokens.light).toHaveProperty('shadows');
    
    // Verify we have essential color tokens
    expect(tokens.light.colors).toHaveProperty('color-primary');
    expect(tokens.light.colors).toHaveProperty('color-bg');
    expect(tokens.light.colors).toHaveProperty('color-surface');
    expect(tokens.light.colors).toHaveProperty('color-text');
    
    // Verify we have typography tokens
    expect(tokens.light.typography).toHaveProperty('font-sans');
    expect(tokens.light.typography).toHaveProperty('font-mono');
    
    // Verify we have radius tokens
    expect(tokens.light.radii).toHaveProperty('radius-sm');
    expect(tokens.light.radii).toHaveProperty('radius-md');
    expect(tokens.light.radii).toHaveProperty('radius-lg');
    
    // Verify dark theme has color overrides
    expect(tokens.dark.colors).toHaveProperty('color-primary');
    expect(tokens.dark.colors).toHaveProperty('color-bg');
    expect(tokens.dark.colors).toHaveProperty('color-surface');
  });
  
  test('tokens.json matches CSS variables count', async () => {
    const tokensPath = path.join(process.cwd(), 'src', 'tokens.json');
    const cssPath = path.join(process.cwd(), 'src', 'styles', 'tokens.css');
    
    const tokensContent = await fs.readFile(tokensPath, 'utf8');
    const cssContent = await fs.readFile(cssPath, 'utf8');
    const tokens = JSON.parse(tokensContent);
    
    // Count CSS variables in :root
    const rootVariables = (cssContent.match(/:root\s*\{[^}]*\}/s)?.[0].match(/--[\w-]+:/g) || []).length;
    
    // Count tokens in light theme
    const lightTokenCount = Object.keys(tokens.light.colors).length + 
                           Object.keys(tokens.light.typography).length + 
                           Object.keys(tokens.light.radii).length + 
                           Object.keys(tokens.light.shadows).length;
    
    expect(lightTokenCount).toBe(rootVariables);
  });
});