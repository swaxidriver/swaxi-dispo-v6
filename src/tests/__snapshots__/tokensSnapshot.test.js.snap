// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Design Tokens Stability tokens.json snapshot remains stable 1`] = `
{
  "dark": {
    "colors": {
      "color-accent": "#27ADE7",
      "color-bg": "#0f172a",
      "color-border": "#334155",
      "color-err": "#f87171",
      "color-muted": "#94a3b8",
      "color-ok": "#16a34a",
      "color-primary": "#8094ff",
      "color-primary-emphasis": "#4f67d2",
      "color-surface": "#1e293b",
      "color-text": "#f1f5f9",
      "color-warn": "#fbbf24",
    },
    "radii": {},
    "shadows": {},
    "spacing": {},
    "typography": {},
  },
  "light": {
    "colors": {
      "color-accent": "#27ADE7",
      "color-bg": "#f8fafc",
      "color-border": "#e2e8f0",
      "color-err": "#dc2626",
      "color-muted": "#64748b",
      "color-ok": "#16a34a",
      "color-primary": "#222F88",
      "color-primary-emphasis": "#1a2566",
      "color-surface": "#ffffff",
      "color-text": "#0f172a",
      "color-warn": "#f59e0b",
    },
    "radii": {
      "radius-lg": "10px",
      "radius-md": "6px",
      "radius-sm": "4px",
    },
    "shadows": {
      "shadow-md": "0 4px 12px -2px rgba(0,0,0,0.12)",
      "shadow-sm": "0 1px 2px rgba(0,0,0,0.06)",
    },
    "spacing": {},
    "typography": {
      "font-mono": "ui-monospace, SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", monospace",
      "font-sans": "Manrope, Inter, system-ui, -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, "Helvetica Neue", Arial, sans-serif",
      "text-lg": "1.125rem",
      "text-md": "1rem",
      "text-sm": "0.875rem",
      "text-xl": "1.25rem",
      "text-xs": "0.75rem",
    },
  },
  "meta": {
    "source": "src/styles/tokens.css",
    "version": "1.0.0",
  },
}
`;
